### PR TITLE
feat(gatsby): support node: protocol when bundling engines

### PR DESCRIPTION
**Pull request by pieh**
## Description

With introduction of `node:` protocol/imports ( https://nodejs.org/api/esm.html#node-imports ) packages begun to use it and this PR adds bundling support for those. Without it we are getting problems like ones described in https://github.com/webpack/webpack/issues/13290

## Related Issues

Fixes https://github.com/gatsbyjs/gatsby/issues/36500

### CODE REVIEW

File: e2e-tests/production-runtime/cypress/integration/ssr.js [hunk 1]
```diff
@@ -173,3 +173,15 @@ describe(`500 status`, () => {
       .should(`exist`)
   })
 })
+
+describe(`"node:" protocol`, () => {
+  it.only(`engines work when bundled code contains node:path import`, () => {
```
Comment (j0sh77) [inline e2e-tests/production-runtime/cypress/integration/ssr.js:178]:
```suggestion
  it(`engines work when bundled code contains node:path import`, () => {
```

**Review by j0sh77 — APPROVED** [review]
Looks good to me minus the one test change

**Review by j0sh77 — CHANGES_REQUESTED** [review]
lol clicked the wrong button

**Review by j0sh77 — APPROVED** [review]
🔥 :shipit: 🔥

**Comment by pieh** [comment]
Ugh, the node version we use for e2e in circleci doesn't support `node:` protocol, so build fails even before compiling / validating engines. Will think on how to address that (without just removing the test heh)

**Comment by pieh** [comment]
I did add hacky polyfill to e2e/prod that allows importing `node:` (we do execute that code during the build so that has to work to test if bundling it works as well), now it should be ready

**Review by marvinjude — APPROVED** [review]

**Pull request merged**